improv: override Tracer auto-capture response/exception via env vars

heitorlessa opened the pull request:
**Issue #, if available:** Feature parity with Java powertools PR https://github.com/awslabs/aws-lambda-powertools-java/pull/249

## Description of changes:

<!--- One or two sentences as a summary of what's being changed -->

Adds env var support to override default behaviour in Tracer of auto-capturing both response and error:

Environment variable | Description
------------------------------------------------- | --------------------------------------------------------------------------------- 
**POWERTOOLS_TRACER_CAPTURE_RESPONSE** | disable auto-capture response as metadata (e.g. `"true", "True", "TRUE"`)
**POWERTOOLS_TRACER_CAPTURE_ERROR** | disable auto-capture error as metadata (e.g. `"true", "True", "TRUE"`)


**Checklist**

<!--- Leave unchecked if your change doesn't seem to apply --> 

* [ ] [Meet tenets criteria](https://awslabs.github.io/aws-lambda-powertools-python/#tenets)
* [x] Update tests
* [x] Update docs
* [x] PR title follows [conventional commit semantics](https://github.com/awslabs/aws-lambda-powertools-python/blob/376ec0a2ac0d2a40e0af5717bef42ff84ca0d1b9/.github/semantic.yml#L2)

## Breaking change checklist

<!--- Ignore if it's not a breaking change -->

**RFC issue #**:

* [ ] Migration process documented
* [ ] Implement warnings (if it can live side by side)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

nmoutschen commented on docs/content/index.mdx:
```diff
@@ -118,15 +118,16 @@ Utility | Description
+**POWERTOOLS_SERVICE_NAME** | Sets service name used for tracing namespace, metrics dimension and structured logging | All | `"service_undefined"`
+**POWERTOOLS_METRICS_NAMESPACE** | Sets namespace used for metrics | [Metrics](./core/metrics) | `None`
+**POWERTOOLS_TRACE_DISABLED** | Disables tracing | [Tracing](./core/tracer) | `false`
+**POWERTOOLS_TRACER_CAPTURE_RESPONSE** | Overrides tracer from auto-capturing response as metadata | [Tracing](./core/tracer) | `true`
```
Is this missing a `POWERTOOLS_TRACER_CAPTURE_ERROR` row?

__NIT__: "Overrides" is not 100% clear to me. Does this means the environment overrides explicit behaviour? Is it in all cases (both true/false) or only in one? Does setting to true means that we override? Since `resolve_env_var_choice` will take the code first then the environment variables, I think we should have something like "Adds the Lambda function response in trace's metadata. Passing it explicitly to the Tracer overrides the environment variable."

heitorlessa replied:
Good call - It's from a previous behaviour I've changed at last minute (explicit wins over env var). Will update it shortly.

nmoutschen commented on aws_lambda_powertools/tracing/tracer.py:
```diff
@@ -449,51 +472,65 @@ async def async_tasks():
+    def _decorate_async_function(
+        self,
+        method: Callable = None,
+        capture_response: Optional[bool] = None,
+        capture_error: Optional[bool] = None,
```
__NOT BLOCKING__: Is there a better way to do this? As time goes and we add new flags/features, this signature will become rather long.

heitorlessa replied:
I couldn't stop to think of that yet, but definitely an area of concern to optimize it for future maintenance. Lemme know if you have ideas - Sounds like a builder or factory pattern could work here.

heitorlessa commented on tests/functional/test_shared_functions.py:
```diff
@@ -0,0 +1,11 @@
+
+
+def test_explicit_wins_over_env_var():
+    choice_env = os.getenv("CHOICE", True)
```
last minute draft that shouldn't be in the commit - will remove it. Any str value for `env` would be the right place, and it's also missing a test for when choice isn't passed.

heitorlessa commented:
Addressed all your comments and some other oversights I've noticed too - Here's the doc too. Over to you @nmoutschen 

![image](https://user-images.githubusercontent.com/3340292/104474134-ff93fa80-55bd-11eb-99c1-132cce7202bc.png)

nmoutschen reviewed (approved):
LGTM
